checker.go

```diff
@@ -34,64 +43,40 @@ type FullScore struct {
 	Negative Score
 }
 
-var lettersAndSpaceOnly = regexp.MustCompile(`[^a-zA-Z ]+`)
-
-// Negativity calculates the negative sentiment of a sentence
-func Negativity(phrase string) Score {
+func calculateScore(phrase string, calcSign sign) Score {
 	var hits float64
 	var words []string
-
-	addPush := func(t string, score float64) {
-		hits -= score
-		words = append(words, t)
-	}
-
+	var count int
+	var lettersAndSpaceOnly = regexp.MustCompile(`[^a-zA-Z ]+`)
+	
 	scanner := bufio.NewScanner(strings.NewReader(strings.ToLower(lettersAndSpaceOnly.ReplaceAllString(phrase, " "))))
 	scanner.Split(bufio.ScanWords)
-
-	var count float64
 	for scanner.Scan() {
 		count++
 		word := scanner.Text()
-		if v, ok := afinn[word]; ok && v < 0 {
-			addPush(word, v)
+		if v, ok := afinn[word]; ok {
+			if (calcSign == positive && v > 0) || (calcSign == negative && v < 0) {
+				hits += v * float64(calcSign)
+				words = append(words, word)
+			}
 		}
 	}
 
 	return Score{
 		Score:       hits,
-		Comparative: hits / count,
+		Comparative: hits / float64(count),
```

Comment:
Why add this extra conversion? Why not work with float64 all the way? Does the increment of a float64 takes longer than an int?